### PR TITLE
Schedule screen

### PR DESCRIPTION
**Pull request by HamzaTrimulabs**

### CODE REVIEW

File: App/@types/index.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  longitude: string;
+  phone: string;
+  primaryColor: string;
+  secondaryColor: string;
+  squareImage: string;
+  state: string;
+  title: string;
+  website: string;
+  zip: string;
+  __typename: string;
```
Comment (alitrimulabs) [inline App/@types/index.ts]: no need for this __typename*

File: App/api/index.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+  ALL_RADIO_QUERY,
+  ALL_PODCASTS_QUERY,
+  SCHEDULE_VIDEO_QUERY,
+  RECENT_RADIO_QUERY,
+  RECENT_PODCAST_QUERY,
+  RECENT_VIDEO_EPISODES_QUERY,
+  RECENT_LIVE_VIDEOS_QUERY,
+  SCHEDULE_LIVE_VIDEO_BY_CHANNEL,
+  NOW_PLAYING_SCHEDULE_VIDEO,
+} from './Queries';
```
Comment (alitrimulabs) [inline App/api/index.js]: use => import Queries from './Queries'

File: App/api/index.js [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+  ALL_RADIO_QUERY,
+  ALL_PODCASTS_QUERY,
+  SCHEDULE_VIDEO_QUERY,
+  RECENT_RADIO_QUERY,
+  RECENT_PODCAST_QUERY,
+  RECENT_VIDEO_EPISODES_QUERY,
+  RECENT_LIVE_VIDEOS_QUERY,
+  SCHEDULE_LIVE_VIDEO_BY_CHANNEL,
+  NOW_PLAYING_SCHEDULE_VIDEO,
+};
```
Comment (alitrimulabs) [inline App/api/index.js]: use => export {Queries}

File: App/components/Calendar/index.tsx [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+import {TrimDate} from '../../util';
+import style from './style';
+
+const Calendar1 = () => {
+  const [isPressed, setIsPressed] = useState(false);
+
+  const toggleOverlay = () => {
+    setIsPressed(!isPressed);
+  };
+  console.log(isPressed);
```
Comment (alitrimulabs) [inline App/components/Calendar/index.tsx]: please do not commit console's

File: App/components/Calendar/index.tsx [hunk 5]
```diff
@@ -0,0 +1,38 @@
+import React, {useEffect, useState} from 'react';
+import {Text, View, TouchableOpacity, Modal} from 'react-native';
+
+import FontTelloIcon from '../FontTelloIcon';
+import {Colors} from '../../constants';
+import {Calendar, CalendarList, Agenda} from 'react-native-calendars';
+import {TrimDate} from '../../util';
+import style from './style';
+
+const Calendar1 = () => {
```
Comment (alitrimulabs) [inline App/components/Calendar/index.tsx]: Try to use more defined naming.

File: App/components/ChannelSchedule/index.tsx [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+
+const Calendar = () => {
+  const {data, selectedOrgId} = useContext(MediaContext);
+  const [isPressed, setIsPressed] = useState(false);
+
+  const AddList = () => {
+    return (
+      <RenderList
+        data={data}
+        // iconName={'clock'}
```
Comment (alitrimulabs) [inline App/components/ChannelSchedule/index.tsx]: do not add commented code

File: App/components/ChannelSchedule/index.tsx [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+        onPress={() => console.log('Channel Schedule to RenderList')}
```
Comment (alitrimulabs) [inline App/components/ChannelSchedule/index.tsx]: remove console

File: App/components/ChannelSchedule/index.tsx [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+        imageType={'square'}
+        descriptionType={'address'}
+        listType={'FindMediaCenters'}
+      />
+    );
+  };
+
+  return (
+    <TouchableOpacity
+      onPress={() => (!isPressed ? setIsPressed(true) : setIsPressed(false))}>
```
Comment (alitrimulabs) [inline App/components/ChannelSchedule/index.tsx]: use => setIsPressed(!isPressed)

File: App/components/ChannelSchedule/index.tsx [hunk 9]
```diff
@@ -0,0 +1,52 @@
...
+        <View style={style.clockIconStyle}>
+          <FontTelloIcon name={'clock'} color={Colors.lightBlue} size={20} />
+        </View>
+        <Text style={style.titleStyle} numberOfLines={1}>
+          {getTitle(data, selectedOrgId.toString())}
+        </Text>
+
+        <View style={style.arrowIconStyle}>
+          <FontTelloIcon
+            name={!isPressed ? 'down-open' : 'up-open'}
```
Comment (alitrimulabs) [inline App/components/ChannelSchedule/index.tsx]: remove ! and replace the strings

File: App/components/ChannelSchedule/index.tsx [hunk 10]
```diff
@@ -0,0 +1,52 @@
...
+            color={Colors.lightBlue}
+            size={20}
+          />
+        </View>
+      </View>
+      {isPressed && data ? AddList() : <NotFound typeName={'Schedule'} />}
```
Comment (alitrimulabs) [inline App/components/ChannelSchedule/index.tsx]: rendern components in jsx => '< Addlist  />'

File: App/components/FindCenter/index.js [hunk 11]
```diff
@@ -1,26 +1,45 @@
...
 import Ionicons from 'react-native-vector-icons/Ionicons';
 import {View, Text, Image, FlatList, TouchableOpacity} from 'react-native';
 import style from './style';
+import {getImage} from '../../util';
+import MediaContext from '../../Context/MediaContext';
+import {SetStorage} from '../../util';
+import {Strings} from '../../constants';
+const CenterComponent = ({onPress1, OrgIds, selectedId, onPress2}) => {
+  const {data} = useContext(MediaContext);
+  selectedId && SetStorage(Strings.SelectedOrganization, selectedId);
```
Comment (alitrimulabs) [inline App/components/FindCenter/index.js]: move this in useEffect

File: App/components/OrgHeader/index.tsx [hunk 12]
```diff
@@ -0,0 +1,48 @@
...
+import MediaContext from '../../Context/MediaContext';
+import {getImage, getTitle} from '../../util';
+import FontTelloIcon from '../FontTelloIcon';
+import {Colors} from '../../constants';
+// import AsyncStorage from '@react-native-async-storage/async-storage';
+
+const OrgHeader = () => {
+  const {data, selectedOrgId} = useContext(MediaContext);
+
+  // const getOrgId = async () => {
```
Comment (alitrimulabs) [inline App/components/OrgHeader/index.tsx]: remove commented code

File: App/components/RenderList/index.js [hunk 13]
```diff
@@ -12,50 +12,44 @@ const RenderList = ({
...
+            source={{
+              uri: imageType === 'square' ? item.squareImage : item.image,
+            }}
+          />
+          <View style={style.textWrap}>
+            <Text numberOfLines={1} style={style.titleStyle}>
+              {item.title}
+            </Text>
+            <Text numberOfLines={1} style={style.detailStyle}>
+              {descriptionType === 'address' ? item.address : item.description}
```
Comment (alitrimulabs) [inline App/components/RenderList/index.js]: Do not use magic string's. Use enums.

File: App/components/RenderList/index.js [hunk 14]
```diff
@@ -12,50 +12,44 @@ const RenderList = ({
...
         data={data}
         horizontal={false}
         ListEmptyComponent={<LoadingSpinner />}
         keyExtractor={(pod) => pod.id.toString()}
-        renderItem={({item}) => {
-          return (
-            <TouchableOpacity onPress={() => onPress(item.id)}>
-              <View
-                style={
-                  listType === 'FindMediaCenters'
```
Comment (alitrimulabs) [inline App/components/RenderList/index.js:29]: again magic string

File: App/components/RenderList/index.js [hunk 15]
```diff
@@ -12,50 +12,44 @@ const RenderList = ({
...
-                    // change in image
-                    uri: imageType === 'square' ? item.squareImage : item.image,
-                  }}
-                />
-                <View style={style.textWrap}>
-                  <Text numberOfLines={1} style={style.titleStyle}>
-                    {item.title}
-                  </Text>
-                  <Text numberOfLines={1} style={style.detailStyle}>
-                    {/* change in address */}
```
Comment (alitrimulabs) [inline App/components/RenderList/index.js:46]: no need for these descriptive comments

File: App/components/RenderLive/index.js [hunk 16]
```diff
@@ -4,6 +4,32 @@ import style from './style';
 import LoadingSpinner from '../LoadingSpinner';
 
 const RenderLive = ({data, channelName, thumbImage, onPress}) => {
+  const renderLive = (item) => {
```
Comment (alitrimulabs) [inline App/components/RenderLive/index.js:8]: renderEpisodes component have many similarity with this comp. can you merge both into one

File: App/navigation/apolloClient/index.js [hunk 17]
```diff
@@ -0,0 +1,8 @@
+import {ApolloClient, InMemoryCache} from '@apollo/client';
```
Comment (alitrimulabs) [inline App/navigation/apolloClient/index.js]: add this client file in api folder.

File: App/navigation/index.js [hunk 18]
```diff
@@ -0,0 +1,95 @@
...
+          if (route.name === 'Home') {
+            iconName = focused ? 'home' : 'home';
+          } else if (route.name === 'Video') {
+            iconName = focused ? 'videocam-1' : 'videocam-1';
+          } else if (route.name === 'Radio') {
+            iconName = focused ? 'signal' : 'signal';
+          } else if (route.name === 'Schedule') {
+            iconName = focused ? 'calendar-empty' : 'calendar-empty';
+          }
+
```
Comment (alitrimulabs) [inline App/navigation/index.js]: move these icons to tab screen tags

File: App/screens/Home/index.js [hunk 19]
```diff
@@ -0,0 +1,196 @@
...
+
+    if (!exist && catId) {
+      const newArr = [];
+      newArr.push(catId);
+      idArr && idArr[0] && newArr.push(idArr[0]);
+      idArr && idArr[1] && newArr.push(idArr[1]);
+
+      setSelectedOrgs(newArr);
+      SetStorage(Strings.RecentlyViewed, newArr);
+      console.log('new Arr is =', newArr);
```
Comment (alitrimulabs) [inline App/screens/Home/index.js]: remove console

File: App/screens/Home/index.js [hunk 20]
```diff
@@ -0,0 +1,196 @@
...
+    }
+  };
+  const getFirstVal = async () => {
+    const idArr = await GetStorage(Strings.RecentlyViewed);
+    setSelectedOrgs(idArr);
+  };
+  useEffect(() => {
+    getFirstVal();
+  });
```
Comment (alitrimulabs) [inline App/screens/Home/index.js]: add [ ]

File: App/util/index.ts [hunk 21]
```diff
@@ -0,0 +1,76 @@
...
+
+const RemoveItem = async (key: string) => {
+  try {
+    AsyncStorage.removeItem(key);
+  } catch (error) {
+    console.log('Error Removing Item', error);
+  }
+};
+
+const getImage = (obj: MediaArr, id: string) => {
```
Comment (alitrimulabs) [inline App/util/index.ts]: rename obj to proper name Like medioObjectList or array, and they Type MediaArr should be MediaObj[ ].

File: App/util/index.ts [hunk 22]
```diff
@@ -0,0 +1,76 @@
...
+    }
+    return second;
+  }
+  return desc;
+};
+
+const TrimDate = () => {
+  let trim = Date().split(' ', 4);
+  return trim[0] + 'day ' + trim[1] + ' ' + trim[2] + ', ' + trim[3];
+};
```
Comment (alitrimulabs) [inline App/util/index.ts]: use moment library for date formatting

File: rn-cli.config.js [hunk 23]
```diff
@@ -0,0 +1,8 @@
+module.exports = {
+  getTransformModulePath() {
+    return require.resolve('react-native-typescript-transformer');
+  },
+  getSourceExts() {
+    return ['ts', 'tsx'];
+  },
+};
```
Comment (alitrimulabs) [inline rn-cli.config.js]: for what purpose this code is added?

File: package.json [hunk 24]
```diff
@@ -16,35 +16,64 @@
...
+    "transform": {
+      "^.+\\.(js)$": "<rootDir>/node_modules/babel-jest",
+      "\\.(ts|tsx)$": "<rootDir>/node_modules/ts-jest/preprocessor.js"
+    },
+    "testRegex": "(/__tests__/.*|\\.(test|spec))\\.(ts|tsx|js)$",
+    "testPathIgnorePatterns": [
+      "\\.snap$",
+      "<rootDir>/node_modules/"
+    ],
+    "cacheDirectory": ".jest/cache"
```
Comment (alitrimulabs) [inline package.json]: for what purpose this code is added?

File: App/navigation/index.js (hunk 18 above)
Comment (HamzaTrimulabs) [reply App/navigation/index.js]: I need to discuss these again for better understanding 
Secondly, I have replaced these strings with enums

File: package.json (hunk 24 above)
Comment (HamzaTrimulabs) [reply package.json]: removed code

File: rn-cli.config.js (hunk 23 above)
Comment (HamzaTrimulabs) [reply rn-cli.config.js]: removed this file

File: App/components/CalendarTab/index.tsx [hunk 25]
```diff
@@ -0,0 +1,56 @@
...
+          <Text style={style.dateStyle}>{TrimDate()}</Text>
+        </View>
+      </TouchableOpacity>
+      <Modal animationType="slide" transparent={true} visible={modalVisible}>
+        <View style={style.centeredView}>
+          <View style={style.modalView}>
+            <Text style={style.modalText}>Select event date</Text>
+            <Calendar />
+            <TouchableHighlight
+              style={{...style.openButton, backgroundColor: '#2196F3'}}
```
Comment (alitrimulabs) [inline App/components/CalendarTab/index.tsx]: use color codes from constants

File: App/components/ScheduleVideoList/index.js [hunk 26]
```diff
@@ -0,0 +1,42 @@
...
+    <View>
+      {scheduleList.length ? (
+        <FlatList
+          data={scheduleList}
+          keyExtractor={(element) => element.id}
+          ListEmptyComponent={<LoadingSpinner />}
+          renderItem={({item}) => (
+            <View style={style.parentStyle}>
+              <View style={style.timeStyle}>
+                <Text numberOfLines={2}>{time}</Text>
```
Comment (alitrimulabs) [inline App/components/ScheduleVideoList/index.js]: As time is a constant you can put the string directly here

File: package.json [hunk 27]
```diff
@@ -20,15 +20,18 @@
     "apollo-client-preset": "^1.0.8",
     "graphql": "^15.4.0",
     "graphql-tag": "^2.11.0",
+    "moment": "^2.29.1",
     "react": "16.13.1",
     "react-apollo": "^3.1.5",
+    "react-moment": "^1.1.1",
```
Comment (alitrimulabs) [inline package.json]: I think this lib is not neccessary

File: App/constants/strings.js [hunk 28]
```diff
@@ -1,6 +1,10 @@
 const Strings = {
   SelectedOrganization: 'SelectedOrganization',
   RecentlyViewed: 'id',
+  CurrentSelectedMediaId: 'CurrentSelectedMediaId',
+  Address: 'address',
+  FindMediaCenters: 'FindMediaCenters',
+  Square: 'square',
```
Comment (alitrimulabs) [inline App/constants/strings.js]: If a string is being used just once you can declare it in that file. like => const SQUARE = 'square' and use.

File: App/components/CalendarTab/style.ts [hunk 29]
```diff
@@ -0,0 +1,61 @@
...
+    shadowOffset: {
+      width: 0,
+      height: 2,
+    },
+    shadowOpacity: 0.25,
+    shadowRadius: 3.84,
+    elevation: 5,
+  },
+  openButton: {
+    backgroundColor: '#F194FF',
```
Comment (alitrimulabs) [inline App/components/CalendarTab/style.ts]: seeing a lot or color code string's being used. Put them in constants.

File: App/components/DropDownChannel/index.tsx [hunk 30]
```diff
@@ -0,0 +1,21 @@
...
+  return (
+    <FlatList
+      data={scheduleList}
+      keyExtractor={(element) => element.id}
+      ListEmptyComponent={<NotFound typeName="Channels" />}
+      renderItem={({item}) => (
+        <View style={style.popUpList}>
+          <Text style={style.popUpStyle} numberOfLines={1}>
+            {item.stations}
+          </Text>
```
Comment (alitrimulabs) [inline App/components/DropDownChannel/index.tsx:25]: extract this renderItem

File: App/components/ChannelSchedule/index.tsx [hunk 31]
```diff
@@ -1,52 +1,105 @@
...
-        descriptionType={'address'}
-        listType={'FindMediaCenters'}
-      />
-    );
+  const getSelectedMedia = async () => {
+    const val = await AsyncStorage.getItem(Strings.CurrentSelectedMediaId);
+    return setCurrentMedia(val);
   };
 
+  const {data: ScheduleData} = useQuery(api.SCHEDULE_VIDEO_QUERY, {
```
Comment (alitrimulabs) [inline App/components/ChannelSchedule/index.tsx]: use Queries.

File: App/components/RenderLive/index.js (hunk 16 above)
Comment (HamzaTrimulabs) [reply App/components/RenderLive/index.js:8]: done

File: package.json (hunk 27 above)
Comment (HamzaTrimulabs) [reply package.json]: removed

File: App/components/CalendarTab/index.tsx [hunk 32]
```diff
@@ -1,20 +1,21 @@
...
 import {Colors} from '../../constants';
-import {Calendar, CalendarList, Agenda} from 'react-native-calendars';
-import {TrimDate} from '../../util';
+import DateTimePickerModal from 'react-native-modal-datetime-picker';
 import style from './style';
 
 const CalendarTab = () => {
-  const [modalVisible, setModalVisible] = useState(false);
+  const [modalVisible, setModalVisible] = useState<boolean>(false);
+  const moment = require('moment');
```
Comment (alitrimulabs) [inline App/components/CalendarTab/index.tsx]: use ES6 syntax

File: App/components/CalendarTab/index.tsx [hunk 33]
```diff
@@ -31,24 +32,18 @@ const CalendarTab = () => {
             />
           </View>
 
-          <Text style={style.dateStyle}>{TrimDate()}</Text>
+          <Text style={style.dateStyle}>
+            {today.format('dddd MMMM D , YYYY')}
```
Comment (alitrimulabs) [inline App/components/CalendarTab/index.tsx]: make this moment.format() directly no need to use const declaration.

File: App/components/ChannelSchedule/index.tsx [hunk 34]
```diff
@@ -23,29 +23,61 @@ import AsyncStorage from '@react-native-async-storage/async-storage';
...
+      },
+    },
+  );
   useEffect(() => {
     if (ScheduleData) {
       setScheduleList(ScheduleData.liveVideosByOrganization);
     }
-  }, [ScheduleData]);
+    if (VideosByChannelData) {
+      console.log(VideosByChannelData.liveVideoByChannel.nodes);
```
Comment (alitrimulabs) [inline App/components/ChannelSchedule/index.tsx]: no console's please

File: App/components/DropDownChannel/index.tsx [hunk 35]
```diff
@@ -1,20 +1,30 @@
...
+      </TouchableOpacity>
+    );
+  };
+
+  return (
+    <FlatList
+      data={scheduleList}
+      keyExtractor={(element) => element.id}
+      ListEmptyComponent={<NotFound typeName="Channels" />}
+      renderItem={({item}) => showList(item)}
```
Comment (alitrimulabs) [inline App/components/DropDownChannel/index.tsx]: showList is not a good naming convention for this purpose.

File: App/components/Header/style.js [hunk 36]
```diff
@@ -9,7 +9,16 @@ export default StyleSheet.create({
     alignItems: 'center',
     flexDirection: 'row',
     justifyContent: 'center',
-    flex: 1,
+    height: 50,
+    backgroundColor: Colors.white,
+    shadowColor: '#000',
```
Comment (alitrimulabs) [inline App/components/Header/style.js]: magic string

File: App/Context/MediaContext.js [hunk 37]
```diff
@@ -2,7 +2,6 @@ import React from 'react';
 
 const MediaContext = React.createContext({
   data: [],
-  id: Number,
```
Comment (alitrimulabs) [inline App/Context/MediaContext.js]: use smallcase e,g. number, string.

File: App/Context/MediaProvider.js [hunk 38]
```diff
@@ -4,9 +4,17 @@ import MediaContext from './MediaContext';
 const MediaProvider = ({children}) => {
   const [data, setData] = useState([]);
   const [selectedOrgId, setSelectedOrgId] = useState(1);
+  const [selectedDate, setSelectedDate] = useState(new Date().toDateString());
```
Comment (alitrimulabs) [inline App/Context/MediaProvider.js]: put new date in a variable.

File: App/components/CalendarTab/index.tsx [hunk 39]
```diff
@@ -15,6 +17,10 @@ const CalendarTab = () => {
   };
   const handleConfirm = (date: Date) => {
     hideDatePicker();
+
+    let myDate = new Date(date);
```
Comment (alitrimulabs) [inline App/components/CalendarTab/index.tsx]: if it's value is not updating then use const.

File: App/components/DropDownChannel/index.tsx [hunk 40]
```diff
@@ -22,9 +31,9 @@ const DropDownChannel = ({getChannelId, scheduleList}) => {
   return (
     <FlatList
       data={scheduleList}
-      keyExtractor={(element) => element.id}
+      keyExtractor={(element) => element.id.toString()}
```
Comment (alitrimulabs) [inline App/components/DropDownChannel/index.tsx]: Prefer to use item keyword.

File: App/navigation/AppTabNavigation.js [hunk 41]
```diff
@@ -0,0 +1,80 @@
+import React from 'react';
+import {createBottomTabNavigator} from '@react-navigation/bottom-tabs';
+import {HomeScreen, VideoScreen, RadioScreen, ScheduleScreen} from '../screens';
+import {Colors, Routes, Icons} from '../constants';
+import FontTelloIcon from '../components/FontTelloIcon';
+
+const MyTabNavigator = () => {
+  const Tab = createBottomTabNavigator();
+  const IconSize = 30;
```
Comment (alitrimulabs) [inline App/navigation/AppTabNavigation.js]: put this in constants and make diff sizes like iconSmall = 20, iconLarge = 40 etc

**Pull request merged**